extensions/barrage/src/main/java/io/deephaven/extensions/barrage/table/BarrageTable.java

```diff
@@ -424,18 +478,26 @@ private void freeRows(final RowSet rowsToFree) {
         }
 
         // Note: these are NOT OrderedRowKeys until after the call to .sort()
-        try (final WritableLongChunk<OrderedRowKeys> redirectedRows =
-                WritableLongChunk.makeWritableChunk(rowsToFree.intSize("BarrageTable"))) {
-            redirectedRows.setSize(0);
+        final int chunkSize = (int) Math.min(rowsToFree.size(), TWO_DIMENSIONAL_COLUMN_SOURCE_THRESHOLD);
```

Comment:
It is probably ever-so-slightly better to use a pooled chunk here. I recommend swapping the max with `1 << ChunkPoolConstants.LARGEST_POOLED_CHUNK_LOG2_CAPACITY` (as you already do in the add batch).